File: UVA/AOAPC-BAC/Ch3/Example/UVA-1584_Circular Sequence.py
"""
    String
    tags: string, 紫書-Ch3
"""
t = int(input())

for _ in range(t):
    s = input()
    n = len(s)
    ans = s
    for i in range(1, n):
        cur = s[i:] + s[:i]
        if cur < ans:
            ans = cur
    print(ans)

File: UVA/AOAPC-BAC/Ch3/Example/UVA-1584_Circular Sequence.cpp
/*
    String
    tags: string, 紫書-Ch3
*/
#include <bits/stdc++.h>
using namespace std;
const int N = 1000;
#define endl '\n'

int main() {
    int t;
    cin >> t;
    string s, ans, cur;
    while (t--){
        cin >> s;
        ans = s;
        for (int i = 1; i < s.size(); i++){
            cur = s.substr(i) + s.substr(0, i);
            if (cur < ans){
                ans = cur;
            }
        }
        cout << ans << endl;
    }
    return 0;
}